File: near/src/main/java/com/adroitandroid/near/discovery/client/BroadcastThread.java
package com.adroitandroid.near.discovery.client;

import android.os.Handler;
import android.os.HandlerThread;

public class BroadcastThread extends HandlerThread {
    private final String mHostName;
    private Handler mHandler;
    private final long mBroadcastInterval;

    BroadcastThread(String hostName, long broadcastInterval) {
        super("Broadcaster");
        mHostName = hostName;
        mBroadcastInterval = broadcastInterval;
    }

    @Override
    protected void onLooperPrepared() {
        mHandler = new BroadcastHandler(getLooper(), mHostName, mBroadcastInterval);
        broadcast();
    }

    void broadcast() {
        mHandler.sendEmptyMessage(BroadcastHandler.REPEAT_BROADCAST);
    }

    void stopBroadcast() {
        if (isAlive()) {
            mHandler.sendEmptyMessage(BroadcastHandler.STOP_BROADCAST);
            quitSafely();
        }
    }
}

File: near/src/main/java/com/adroitandroid/near/discovery/client/BroadcastThread.kt
package com.adroitandroid.near.discovery.client

import android.os.Handler
import android.os.HandlerThread
import org.json.JSONObject

class BroadcastThread internal constructor(private val mHostJson: JSONObject,
                                           private val mBroadcastInterval: Long,
                                           private val port: Int) : HandlerThread("Broadcaster") {
    private lateinit var mHandler: Handler
    override fun onLooperPrepared() {
        mHandler = BroadcastHandler(looper, mHostJson, mBroadcastInterval, port)
        broadcast()
    }

    fun broadcast() {
        mHandler.sendEmptyMessage(BroadcastHandler.REPEAT_BROADCAST)
    }

    fun stopBroadcast() {
        if (isAlive) {
            mHandler.sendEmptyMessage(BroadcastHandler.STOP_BROADCAST)
            quitSafely()
        }
    }

}